model/storage-private/src/main/java/org/keycloak/models/session/PersistentAuthenticatedClientSessionAdapter.java

```diff
@@ -302,10 +282,6 @@ protected static class PersistentClientSessionData {
         private Set<String> protocolMappers;
         @JsonProperty("roles")
         private Set<String> roles;
-        @JsonProperty("currentRefreshToken")
```

Comment:
Take care with this. It seems that this is serialized inside the database and if properties are removed an exception is thrown when an offline session is restored from database. Check the comment just before this point. I just added this line:

```java
    @JsonIgnoreProperties(ignoreUnknown = true)
    protected static class PersistentClientSessionData {
```

It seems that now in quarkus it works OK.

Reply:
@rmartinc Good catch! +1 for adding the test.

There is already `AbstractMigrationTest.testOfflineTokenLogin()`, but looks that this one is not tested with "revoke refresh token" enabled? Maybe we can just update existing realm(s) JSON to use "revoke refresh token" though? At least for the last migration (probably `migration-realm-19.0.3.json`) to make sure it is tested in GH actions.

Reply:
Thanks @rmartinc  and @mposolda ! Running migration tests with `revokeRefreshToken` enabled in `migration-realm-24.0.4.json`, I was unable to reproduce the issue. After some investigation, it seems that `currentRefreshToken` and `currentRefreshTokenUseCount` of `PersistentClientSessionData` are not being saved in the database. To be sure I have added `@JsonIgnoreProperties(ignoreUnknown = true)` to `PersistentClientSessionData` and enabled `revokeRefreshToken` in `migration-realm-24.0.4.json`.